### PR TITLE
Remove bootstrap server and enclave

### PR DESCRIPTION
**Pull request by ThibsG**

### CODE REVIEW

File: crate/server/README.md [hunk 1]
```diff
@@ -7,7 +7,6 @@ The KMS server provides several features which can be enabled at compilation tim
 | Feature  | Description                                                                                                       | Staging | Prod 🔥 |
 |----------|-------------------------------------------------------------------------------------------------------------------|---------|---------|
 | auth     | Enable authentication. If disabled, multi-user is not supported                                                   | ✅       | ✅       |
-| enclave  | Enable the ability to run inside an enclave                                                                       | ✅       | ✅       |
 | https    | Enable https in the KMS in order to encrypt query between client and the KMS. If disabled, it uses http           | ✅       | ✅       |
 | insecure | Do not verify auth0 token expiration date and https ssl is self-signed (to avoid to be banned by letsencrypt)     | ✅       |         |
```
Comment (heavenboy8) [inline crate/server/README.md:11]:
```suggestion
| insecure | Do not verify auth0 token expiration date     | ✅       |         |
```

File: crate/server/README.md [hunk 2]
```diff
@@ -33,7 +32,6 @@ For staging environment, you can use `--features=staging --no-default-features`.
 
 - to not verify the expiration of OAuth2 tokens if `KMS_JWT_ISSUER_URI` is set.
 - to use HTTPS connection with unsecure SSL certificates (it will play anyway all the process to get a valid certificates and starts a HTTPS server)
```
Comment (heavenboy8) [inline crate/server/README.md:34]:
```suggestion
```

Comment (heavenboy8) [reply crate/server/README.md:34]: Same on the previous line (23/24) : - to use HTTP connection

File: crate/server/README.md [hunk 3]
```diff
@@ -7,7 +7,6 @@ The KMS server provides several features which can be enabled at compilation tim
 | Feature  | Description                                                                                                       | Staging | Prod 🔥 |
 |----------|-------------------------------------------------------------------------------------------------------------------|---------|---------|
 | auth     | Enable authentication. If disabled, multi-user is not supported                                                   | ✅       | ✅       |
-| enclave  | Enable the ability to run inside an enclave                                                                       | ✅       | ✅       |
 | https    | Enable https in the KMS in order to encrypt query between client and the KMS. If disabled, it uses http           | ✅       | ✅       |
```
Comment (heavenboy8) [inline crate/server/README.md:10]:
```suggestion
```

File: crate/server/src/config/command_line/http_config.rs [hunk 4]
```diff
@@ -8,19 +8,15 @@ pub struct HttpConfig {
...
-    ///
-    /// The PKCS#12 can be securely provided via the bootstrap server. Check the documentation.
     #[clap(long, env = "KMS_HTTPS_P12_FILE")]
     pub https_p12_file: Option<PathBuf>,
 
     /// The password to open the PKCS#12 Certificates and Key file
-    ///
-    /// The PKCS#12 password can be securely provided via the bootstrap server. Check the documentation.
     #[clap(long, env = "KMS_HTTPS_P12_PASSWORD", default_value = "")]
     pub https_p12_password: String,
```
Comment (heavenboy8) [inline crate/server/src/config/command_line/http_config.rs:22]: Remove that?

File: crate/server/src/config/params/http_params.rs [hunk 5]
```diff
@@ -1,36 +1,19 @@
...
-    config::{command_line::HttpsCertbotConfig, ClapConfig, WorkspaceConfig},
-    core::certbot::Certbot,
-    result::KResult,
-};
+use crate::{config::ClapConfig, result::KResult};
 
 /// The HTTP parameters of the API server
 pub enum HttpParams {
-    Certbot(Arc<Mutex<Certbot>>),
     Https(ParsedPkcs12_2),
```
Comment (heavenboy8) [inline crate/server/src/config/params/http_params.rs:9]: I think we can remove HTTPS. Will be handle by an nginx?

File: crate/server/src/config/params/server_params.rs [hunk 6]
```diff
@@ -55,20 +49,10 @@ pub struct ServerParams {
     // pub certbot: Option<Arc<Mutex<Certbot>>>,
     pub http_params: HttpParams,
 
-    /// The tee parameters when running inside a tee
-    pub tee_params: TeeParams,
-
     /// The certificate used to verify the client TLS certificates
     /// used for authentication
     pub verify_cert: Option<X509>,
```
Comment (heavenboy8) [inline crate/server/src/config/params/server_params.rs:54]:
```suggestion
    pub client_cert: Option<X509>,
```

**Comment by heavenboy8** [comment]
@ThibsG  actually you can remvoe the rust feature "https". If p12 enables HTTPS KMS, if not allow http

**Review by heavenboy8 — APPROVED** [review]

**Review by Manuthor — APPROVED** [review]

**Pull request merged**